components/more_direct_channels/index.ts

```diff
@@ -99,7 +99,7 @@ const filterGroupChannels = memoizeResult((channels: Array<{profiles: UserProfil
     });
 });
 
-const filterDirectChannels = memoizeResult((channels: Channel[], userId: string) => {
+const filterDirectChannels = memoizeResult((channels: Record<string, Channel>, userId: string) => {
```

Comment:
This type was incorrect before, but it didn't matter until I updated the dependencies in mattermost-redux